lib/validator/validator.go

```diff
@@ -92,16 +100,26 @@ func init() {
 	// Register field validators.
 	registerFieldValidator("action", validateAction)
 	registerFieldValidator("interface", validateInterface)
+	registerFieldValidator("datastoreType", validateDatastoreType)
 	registerFieldValidator("backendaction", validateBackendAction)
 	registerFieldValidator("name", validateName)
 	registerFieldValidator("namespacedname", validateNamespacedName)
 	registerFieldValidator("selector", validateSelector)
 	registerFieldValidator("tag", validateTag)
 	registerFieldValidator("labels", validateLabels)
+	registerFieldValidator("labelsToApply", validateLabelsToApply)
+	registerFieldValidator("labelsToApply", validateLabelsToApply)
 	registerFieldValidator("scopeglobalornode", validateScopeGlobalOrNode)
 	registerFieldValidator("ipversion", validateIPVersion)
 	registerFieldValidator("ipipmode", validateIPIPMode)
 	registerFieldValidator("policytype", validatePolicyType)
+	registerFieldValidator("bgpLogLevel", validateBGPLogLevel)
+	registerFieldValidator("felixLogLevel", validateFelixLogLevel)
+	registerFieldValidator("dropacceptreturn", validateFelixEtoHAction)
+	registerFieldValidator("acceptreturn", validateAcceptReturn)
+	registerFieldValidator("ipv4", validateIPv4Address)
```

Comment:
I thought the validator had built-ins for ip, ipv4 and ipv6?

Reply:
same should be true for cidrs I think (although if we allow IPs interchangeably with CIDRs then I daresay we'd need to handle that).

Reply:
We use IPv4 and IPv4+CIDR interchangeably which `validator.IPv4` sadly does not support. Let's stick with our home-grown validators since they're efficient, minimal and reflect what the backend is using as well.

Reply:
Yup that's fine - but perhaps we could rename it - cidrOrIP or something similar

Reply:
how is: `validateIPv4orCIDRAddress` for example?
Should really be `validateIPv4OrCIDRAddress` but that kind of looks like 40.